scripts/devservices-healthcheck.sh

```diff
@@ -13,21 +13,47 @@
 #       specify distinct retries and other parameters.
 #       would also be nice to have a single ^C work.
 
-try=1
-to=3
-start=0
-delay=5
-
-sleep "$start"
-while (( $try <= $to )); do
-    echo "Checking health of postgres (try ${try} of ${to})..."
-    if docker exec sentry_postgres pg_isready -U postgres; then
-        break
-    fi
-    if (( $try == $to )); then
-        echo "Exceeded retries."
-        exit 1
-    fi
-    try=$(($try + 1))
-    sleep "$delay"
-done
+
+# Accepts following positional arguments:
+# $1 - how many times to try
+# $2 - delay between tries
+# $3 - the service (docker container) name
+# $4 - command to run inside the docker container
+function health_check {
+  local try=1
+  local to=$1
+  local start=0
+  local delay=$2
+  local service="$3"
+  local cmd="$4"
+
+
+  sleep "$start"
+  echo "Running '$cmd' in '$service'"
+  while (( $try <= $to )); do
+      echo "Checking '$service' (try ${try} of ${to})..."
+      if docker exec $service $cmd; then
+          return 0
+      fi
+
+      if (( $try == $to )); then
+          echo "Exceeded retries for '$service'"
+          return 1
+      fi
+      try=$(($try + 1))
+      sleep "$delay"
+  done
+}
+
+# Check the postgres statu
+health_check 3 5 "sentry_postgres" "pg_isready -U postgres"
+[[ $? == 1 ]] && exit 1
+
+# Check the kafka cluster status
+if [ "$NEED_KAFKA" = "true" ]; then
+  health_check 3 5 "sentry_kafka" "kafka-topics --zookeeper 127.0.0.1:2181 --list"
+  [[ $? == 1 ]] && exit 1
```

Comment:
nit: I get that returning a value from `health_check` is more modular, but for convenience wouldn't it be nicer to call the `exit 1` inside the function?

Reply:
Yeah, you're right.  It might be clearer and shorter if we exit in the function itself. I'll change it.